app/controllers/support_interface/support_interface_controller.rb

```diff
@@ -16,5 +18,25 @@ def protect_with_basic_auth
     def render_404
       render 'errors/not_found', status: :not_found
     end
+
+    def current_support_user
+      SupportUser.load_from_session(session)
```

Comment:
Given that there's nothing to be done if the support user is not authorized, what would be the effect of making an  returning `nil` if the user is unauthorized?. Maybe `load_from_session` itself could return nil? Not sure of the right place to cut it.

Reply:
This bit does feel messy. The tricky bit is dealing with users that are logged in but don't have authorisation. We can either leave them logged in (in a kind of limbo state) and tell them they don't have authorisation or immediately end their session. I went with the first option mainly because it mirrors what we do in the provider interface but also to avoid people thinking they've mistyped a password or something when it's just a permission thing.